torch/_decomp/decompositions.py

```diff
@@ -4034,6 +4035,23 @@ def inplace_op(*args, **kwargs):
     return inplace_op
 
 
+@out_wrapper()
+@pw_cast_for_opmath
+@register_decomposition([aten.baddbmm])
```

Comment:
You're registering the decomposition before applying the `pw_cast_for_opmath` wrapper. This PR shouldn't have been merged with the tests still skipped.